lib/demo_web/controllers/user_auth.ex

```diff
@@ -0,0 +1,116 @@
+defmodule DemoWeb.UserAuth do
+  import Plug.Conn
+  import Phoenix.Controller
+
+  alias Demo.Accounts
+  alias DemoWeb.Router.Helpers, as: Routes
+
+  # Make the remember me cookie valid for 60 days.
+  # If you want bump or reduce this value, also change
+  # the token expiry itself in UserToken.
+  @max_age 60 * 60 * 24 * 60
+  @remember_me_cookie "user_remember_me"
+
+  @doc """
+  Logs the user in.
+
+  It deletes the CSRF token and renews the session
+  to avoid fixation attacks.
+  """
+  def login_user(conn, user, params \\ %{}) do
+    Plug.CSRFProtection.delete_csrf_token()
+    token = Accounts.generate_session_token(user)
+
+    user_return_to = get_session(conn, :user_return_to)
+    delete_session(conn, :user_return_to)
```

Comment:
Consider defaulting to a full `clear_session` to prevent any kind of fixation. There may be cases where people will want to opt out of a full clear, but that would presumably be the minority of cases.